### ISSUE TITLE
Unable to create two gstore instances bound to two separate namespaces

### ISSUE THREAD

**Issue by jainanshul** [sebelga/gstore-node#59]
STR

```
const gstore = require('gstore-node');
const datastore1 = require('@google-cloud/datastore')({
    projectId: 'my-google-project-id',
   namespace: `1`
});

// Then connect gstore to the datastore
gstore.connect(datastore1);

const datastore2 = require('@google-cloud/datastore')({
    projectId: 'my-google-project-id',
   namespace: `2`
});
gstore.connect(datastore2);

```

The second datastore `datastore2` overwrites the first one and therefore calls to `delete` operation, for example, on `datastore1` are done on `namespace2` instead and therefore fails. The issue is that we export `new Gstore ` at https://github.com/sebelga/gstore-node/blob/master/lib/index.js#L135 instead of ust `Gstore` so there is no way to create two separate instances of `Gstore`

**Comment by sebelga** [comment]
Hi,
What is the use case of having several instances? Is it just to have different namespace? In that case all the methods accept a namespace argument that should allow switching namespace anytime.

```js

const namespace = require('./config').namespace1; // for example

Model.get(123, null, namespace).then( ... );
```
Would that not work for you?

**Comment by jainanshul** [comment]
That should work, however 1) It seems unnecessary to pass the namespace when you already told `'@google-cloud/datastore'` what your namespace is 2) It is inconsistent behavior since you don't really have to pass the namespace unless you get in this situation so prone to bugs.

**Comment by sebelga** [comment]
Ok I understand you have several datastore instances running. I never had this situation as I only configure the datastore once when my server start.

By curiosity, what is your use case ? As for me the purpose of the namespace is mainly to differentiate the environments of the app. Another use case would be to have one namespace by "business client", then as it is dynamic, you can not create "instances" of datastore ahead... So I am curious :)

**Comment by jainanshul** [comment]
I have multiple namespaces to help isolate different sources/types of data. Each datastore instance is associated with a given namespace and was hoping I could use gstore-node instance per datastore client per namespace so rest of the code I have doesn't need to pass in the namespaces to interact with datastore and simply operates on a specific gstore instance.

Your suggestion above to pass in namespace works but I think having a global or a singleton instance of `gstore-node` when it is in fact wrapping a unique datastore instance is limiting. Not a big deal as I could work around it (thanks for your suggestion!)

**Comment by sebelga** [comment]
I see. I will have a look into it. For now it needs to be singleton to ensure that Models are defined only once. That's why you can do this anywhere to retrieve a Model:

```js
const gstore = require('gstore-node');
const MyModel = gstore.model('MyModel');
```
Having several instances would mean to store the Models differently.

**Comment by 6uliver** [comment]
I did the same workaround, I'm passing in namespace parameter for some call. 

I have another use case for namespaces: I have a legacy datastore schema and I'm migrating to a new schema. I've decided to separate the different versions with namespaces, v1, v1.1, v2 etc so we can use the old and new one at the same time without interference.

IMO this enhancement would help to write data migration code easier for our use case.

**Comment by sebelga** [comment]
Thanks for the feedback. I will look into a solution for this.

**Comment by sebelga** [comment]
Hi,
I just released v2.0.0 that support multiple instances of gstore. I still have to update the docs but you can see how to do it in the release notes.
Let me know if there is any issue. thanks!

**Comment by 6uliver** [comment]
Thank you very much!
I reviewed the commit and LGTM. In the next days I will try it.
